src/nbl/video/IOpenGL_LogicalDevice.h

```diff
@@ -444,10 +444,12 @@ class IOpenGL_LogicalDevice : public ILogicalDevice, protected impl::IOpenGL_Log
         template <typename RequestParams>
         void waitForRequestCompletion(SRequest& req)
         {
-            auto lk = req.wait_for_result();
+            req.wait_ready();
```

Comment:
did you make sure no logical device functions use request api directly? (and not waitForRequestCompletion) From what i see user is always required to call wait_ready() and then discard_storage(). 
btw: nabla's interfaces IAsyncQueueDispatcher and Cancellable one dont provide `waitForRequestCompletion` so it must be written by user (using request api directly), i hope it's not an issue

Reply:
yes.